rfcs/2024-08-21-motion-presets.md

```diff
@@ -0,0 +1,523 @@
+---
+Start Date: 21-08-2024
+RFC PR: (leave this empty if no PR yet)
+Blade Issue: (leave this empty if no issue yet)
+---
+
+# Motion Presets RFC <!-- omit in toc -->
+
+### Table Of Contents <!-- omit in toc -->
+
+- [Summary](#summary)
+- [Basic Example](#basic-example)
+- [Motivation](#motivation)
+- [Detailed Design](#detailed-design)
+- [Drawbacks/Constraints](#drawbacksconstraints)
+- [Alternatives](#alternatives)
+- [Adoption strategy](#adoption-strategy)
+- [How do we educate people?](#how-do-we-educate-people)
+- [Open Questions](#open-questions)
+- [References](#references)
+
+# Summary
+
+# Basic Example
+
+```jsx
+import { Fade, Card, CardBody } from '@razorpay/components/blade';
+
+<Fade isVisible={showCard}>
```

Comment:
How do we pass duration tokens etc?

Reply:
we can't. Which is why we have preset so that we can define durations and easing on preset level and make it consistent

Reply:
> define durations and easing on preset level
Do you mean when we define the tokens?

Reply:
No like internally durations are defined for each preset. E.g Fade has certain given duration that cannot be changed on design as well